### PR TITLE
Update references to shielded transactions

### PR DESCRIPTION
**Pull request by ptagl**
After the development of the removal of shielded transactions #628, docs on the repository have been updated to reflect the current status of Zend and remove references to any feature that is not available anymore.

Some sections have been kept and marked as "deprecated" since they may still be useful somehow after the hard fork activation.

### CODE REVIEW

File: doc/wallet-backup.md [hunk 1]
```diff
@@ -36,7 +36,27 @@ If you generate a new Zcash address, it will not be reflected in the backup file
...
+
+This will return the private key and will not create a new file.
+
+To import a private key for a transparent address, use:
+
+`zcash-cli importprivkey <t-priv-key>`
+
+See the command's help documentation for instructions on fine-tuning the wallet rescan:
+
+`zcash-cli help importprivkey`
```
Comment (cronicc) [inline doc/wallet-backup.md]: zen-cli

File: doc/wallet-backup.md [hunk 2]
```diff
@@ -48,20 +68,14 @@ To import keys into a wallet which were previously exported to a file, use:
 
 `zcash-cli z_importwallet </path/to/exportdir/nameofbackup>`
 
-### Using `z_exportkey`, `z_importkey`, `dumpprivkey` & `importprivkey`
+### Using `z_exportkey` & `z_importkey` (deprecated)
 
 If you prefer to export a single private key for a shielded address, you can use:
 
 `zcash-cli z_exportkey <z-address>`
```
Comment (cronicc) [inline doc/wallet-backup.md]: See above

**Review by cronicc — APPROVED** [review]
Approved, comments non-blocking

File: doc/wallet-backup.md (hunk 1 above)
Comment (ptagl) [reply doc/wallet-backup.md]: Thanks, I've updated all the occurrences!

File: doc/wallet-backup.md (hunk 2 above)
Comment (ptagl) [reply doc/wallet-backup.md]: Done!

**Review by JackPiri — APPROVED** [review]
The PR is fine to me.
Just mentioning in `payment-api.md` there are multiple references to privacy features (z_* RPCs, zaddr, zcash-cli, etc) probably we want to get rid of.
Thanks

**Review by drgora — APPROVED** [review]

File: doc/wallet-backup.md [hunk 3]
```diff
@@ -14,7 +14,7 @@ Note that a backup is a duplicate of data needed to spend ZEC so where you keep
 
 These instructions are specific for the officially supported Zcash Linux client. For backing up with third-party wallets, please consult with user guides or support channels provided for those services.
 
-There are multiple ways to make sure you have at least one other copy of the private keys needed to spend your ZEC and view your shielded ZEC.
+There are multiple ways to make sure you have at least one other copy of the private keys needed to spend your ZEN.
```
Comment (a-petrini) [inline doc/wallet-backup.md:17]: For uniformity, I would change all the other instances of `ZEC` to `ZEN`, and `ZCash` to `Zen`

**Review by a-petrini — APPROVED** [review]
Good for me, I left a minor non/blocking comment

File: doc/wallet-backup.md (hunk 3 above)
Comment (ptagl) [reply doc/wallet-backup.md:17]: Done, thanks!

**Comment by ptagl** [comment]
I've added a couple of commits to address the requested changes, let me know if they look ok.

`payment-api.md` could be extended to add more information (for instance how the `z_mergetoaddress` command works), but if it's not crucial, we may postpone this task to another PR.

**Pull request merged**